src/performance_counters/register_counters.cpp

```diff
@@ -46,6 +150,38 @@ namespace phylanx { namespace performance_counters
             &ir::node_data<double>::move_assignment_count,
             "returns the current value of the move-assignment count of "
                 "any node_data<double>");
+
+
+        namespace et = phylanx::execution_tree;
+
+        // Query the list of existing primitive types
+        et::pattern_list const& pattern_list = et::get_all_known_patterns();
+
+        // Iterate and register a time and count performance counter per each primitive
+        for (auto const& patterns : pattern_list)
+        {
+            // No need to process all entries in the list
+            auto const& pattern = *(patterns.begin());
+            // The name of the primitive
+            std::string const& name = hpx::util::get<0>(pattern);
+            // Register a primitive time performance counter
+            hpx::performance_counters::install_counter_type(
+                "/phylanx/time/primitives/" + name + "/eval",
```

Comment:
May I suggest we stick to the naming conventions as outlined in #96? This would make the new counter names conform to other, already existing counters.

Reply:
The naming convention in #96 is hard to read. e.g Upon seeing a list of performance counters `/phylanx{locality#*/total}/add/count/eval` does not suggest that this performance counter measures something about add primitives. In this respect `/phylanx{locality#N/total}/primitives/<primitive>/count/eval` is better.

Reply:
Ok, but the currently registered name is different from that.

Reply:
Must have been a mistake I amended. That's what it looks like right now.

Reply:
Ok.